### ISSUE TITLE
RuntimeError: Non-empty 4D data tensor expected but got a tensor with sizes [1, 0, 160, 160]

### ISSUE THREAD

**Issue by Dicko87** [abdksyed/DETR#1]
Hi there, when running the model, I ran into the error shown below.
I made the changes you kindly recommended to the engine.py file but still I get this message.
Any help would be great, thank you.
![image](https://user-images.githubusercontent.com/37139958/136270212-63b524b5-3f81-4da1-8d2c-5f61edcca765.png)

I tried the model on a very small dataset originally (without changing the engine.py file) and everything worked okay but as soon as I give the model a larger dataset I run into this problem.

Maybe there's something wrong with the annotations, but I have checked them.

**Comment by abdksyed** [comment]
Have you changed the number of classes, or the newer dataset has different number of classes, especially different class ids??

**Comment by Dicko87** [comment]
Hi there, I have not changed the number of classes but I'm guessing something has been altered along the way in my json files so I am going to try and find out what has changed today. I'll let you know :) thank for your help.

**Comment by abdksyed** [comment]
Yeah, you can programmatically, find if any of the annotations in the JSON has category id outside the category ids present in category field of JSON and some other sanity checks on JSON.
One sanity check which DETR already does is it checks if mask file name is same as images file name for panoptic segmentation.

**Comment by Dicko87** [comment]
Thanks, will keep that in mind. SO I am starting from the beginning and with the older very small dataset, the model runs no problem and I get the following result at the end.
![image](https://user-images.githubusercontent.com/37139958/136351535-55472cc2-235e-406a-b7c3-537bfb0771b2.png)

Just need to find where I've gone wrong when putting my images together etc.

**Comment by Dicko87** [comment]
My first thought is, does the first images 'id' need to be 0 or can it be any number as long as it matches the annotations, because I have noticed that the first id is 566. Maybe I could put them in numerical order to see.

For example, I currently have:
![image](https://user-images.githubusercontent.com/37139958/136353871-b600822a-53d4-4599-ae62-ac185350c8f7.png)

With the first annotation:
![image](https://user-images.githubusercontent.com/37139958/136353940-fac64dc5-e11d-497d-b66c-75558dc235ec.png)

Here the image_id in the annotations matches the image, but also noticed the first index of the annotations is indexed at 0.

**Comment by Dicko87** [comment]
So I spent the morning creating  a script to put the images and annotations in numerical order. I then took a small subset (10 training and 5 validation) images and corresponding jsons, put it through the model and the model trained fine.
I then put the full dataset through (again sorted numerically) and ran into the same error as before.
Either there are problem annotations / images that weren't there in the small subset, or the model does not like the size of the dataset I have given it (2920 train and 730 valid).

**Comment by Dicko87** [comment]
I changed the arguments to run the model for 1 epcoh and batch size of 1 to see what difference that would make and it is seems the model gets stuck on 330/730 test / validation images.
![image](https://user-images.githubusercontent.com/37139958/136388998-56bd628d-c1fe-4fba-8fc0-106602a262e0.png)

**Comment by Dicko87** [comment]
I reduced the number of images, this time 1000 training images with 300 validation images and the model ran ok.
![image](https://user-images.githubusercontent.com/37139958/136393621-3886f445-142b-473a-bb05-afefb2b9ad2d.png)
![image](https://user-images.githubusercontent.com/37139958/136393683-2fa76a56-6922-45c3-9649-b7acb7483b29.png)
I will increase it further again and see where the limit is - very strange.

The following warning pops up every time the model is run, even when it runs with no problem.
![image](https://user-images.githubusercontent.com/37139958/136394076-4959805d-21d0-4127-b3c3-ff360411dd26.png)
I wonder if this has anything to do with it.

**Comment by abdksyed** [comment]
I think there is an issue with the annotations file. If you are comfortable can you give the json file, I can run few sanity checks and see.

And as for the up sampling, it's just a warning, that should not cause any error.

**Comment by Dicko87** [comment]
Thanks very much, well I have been going through it one by one and found some suspect annotations (the ones that caused the model to stop working). I am almost done and if there aren't too many I might just remove the images from the dataset. If I get stuck with it then I can maybe let you do a sanity check if you dont mind - thank you :D

**Comment by Dicko87** [comment]
I managed to remove the dodgey annotations / images (only 4) and now the model is running fine thank you.
I wanted to ask, once the model has finished training, how to I run a prediction on my test image and display the resulting png and mask image (the two of them) ?
